src/library_webgpu.js

```diff
@@ -1256,16 +1256,21 @@ var LibraryWebGPU = {
     {{{ gpu.makeCheckDescriptor('descriptor') }}}
     function makePrimitiveState(rsPtr) {
       if (!rsPtr) return undefined;
-      {{{ gpu.makeCheckDescriptor('rsPtr') }}}
+      {{{ gpu.makeCheck('rsPtr') }}}
+
+      var nextInChainPtr = {{{ makeGetValue('rsPtr', C_STRUCTS.WGPUPrimitiveState.nextInChain, '*') }}};
+      var sType = {{{ gpu.makeGetU32('nextInChainPtr', C_STRUCTS.WGPUChainedStruct.sType) }}};
```

Comment:
nextInChainPtr may be null, so that needs to be handled here.

Also, I think there's only one struct that can be chained off PrimitiveState right now, but officially there can be more things in the chain, so ideally we would traverse the whole thing. HOWEVER, that said, we are planning to get rid of this chained struct and move it into PrimitiveState directly:
https://github.com/webgpu-native/webgpu-headers/issues/212#issuecomment-1682801259
So there is no need to do the full traversal thing right now if we're going to go and delete it soon anyway.

```suggestion
      // TODO: This small hack assumes that there's only one type that can be in the chain of
      // WGPUPrimitiveState. The correct thing would be to traverse the chain, but unclippedDepth
      // is going to move into the core object soon, so we'll just do this for now. See:
      // https://github.com/webgpu-native/webgpu-headers/issues/212#issuecomment-1682801259
      var nextInChainPtr = {{{ makeGetValue('rsPtr', C_STRUCTS.WGPUPrimitiveState.nextInChain, '*') }}};
      var sType = nextInChainPtr ? {{{ gpu.makeGetU32('nextInChainPtr', C_STRUCTS.WGPUChainedStruct.sType) }}} : 0;
```

Reply:
Sweet, didn't know that GitHub had suggestions - incorporated your version!